### PR TITLE
Add metric to track terminating endpoint events

### PR DESCRIPTION
**Pull request by aditighag**
Add a new metric to track terminating endpoint
events received from Kubernetes. This can help
to check if service endpoints can be gracefully
terminated by cilium. Cilium agent will receive
these events (and thereby increment the metric)
only when the Kubernetes feature gate
`EndpointSliceTerminatingCondition` is enabled, and
service endpoint pods are correctly configured with
a graceful termination period.

Signed-off-by: Aditi Ghag <aditi@cilium.io>

### CODE REVIEW

**Review by dylandreimerink — APPROVED** [review]
LGTM

File: Documentation/operations/metrics.rst [hunk 1]
```diff
@@ -362,13 +362,14 @@ Name                                     Labels
...
-======================================== ================================================== ========================================================
+==========================================  ================================================ ========================================================
+Name                                        Labels                                           Description
+=========================================== ===============================================  ========================================================
+``kubernetes_events_received_total``        ``scope``, ``action``, ``validity``, ``equal``   Number of Kubernetes events received
+``kubernetes_events_total``                 ``scope``, ``action``, ``outcome``               Number of Kubernetes events processed
+``k8s_cnp_status_completion_seconds``       ``attempts``, ``outcome``                        Duration in seconds in how long it took to complete a CNP status update
+``k8s_terminating_endpoints_events_total``  ``subsystem``                                    Number of terminating endpoint events received from Kubernetes
```
Comment (sayboras) [inline Documentation/operations/metrics.rst]: I think subsystem (e.g. k8s) is part of the name, but not the label.

File: pkg/k8s/endpoints.go [hunk 2]
```diff
@@ -196,6 +197,7 @@ func ParseEndpointSliceV1Beta1(ep *slim_discovery_v1beta1.EndpointSlice) (Endpoi
 				if option.Config.EnableK8sTerminatingEndpoint {
 					if sub.Conditions.Terminating != nil && *sub.Conditions.Terminating {
 						backend.Terminating = true
+						metrics.TerminatingEndpointsEvents.Inc()
```
Comment (sayboras) [inline pkg/k8s/endpoints.go]: for EndpointSliceV1 as well ? 

https://github.com/cilium/cilium/blob/7e07d55efdd8f721f2caf15fd4212336a6478129/pkg/k8s/endpoints.go#L283

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -1105,6 +1110,17 @@ func CreateConfiguration(metricsEnabled []string) (Configuration, []prometheus.C
 			collectors = append(collectors, KubernetesCNPStatusCompletion)
 			c.KubernetesCNPStatusCompletionEnabled = true
 
+		case Namespace + "_terminating_endpoints_events_total":
+			TerminatingEndpointsEvents = prometheus.NewCounter(prometheus.CounterOpts{
```
Comment (sayboras) [inline pkg/metrics/metrics.go]: Does having nodename as label provide any value ? Just my thinking about having one more dimension for troubleshooting (e.g. problematic node).

**Review by sayboras — APPROVED** [review]
Some minor comments as per below

File: pkg/metrics/metrics.go (hunk 3 above)
Comment (aditighag) [reply pkg/metrics/metrics.go]: Sure, good idea! Thanks!

**Review by joestringer — APPROVED** [review]
LGTM, thanks!

**Comment by aditighag** [comment]
Fixed the table formatting issue in the doc.

**Pull request merged**